packages/combobox/src/types.ts

```diff
@@ -0,0 +1,33 @@
+import { PlateEditor, TElement } from '@udecode/plate-common';
+
+export interface TriggerComboboxPlugin {
+  combobox?: {
+    trigger?: string;
+    triggerPreviousCharPattern?: RegExp;
+    query?: (editor: PlateEditor) => boolean;
+    createInputNode?: () => TElement;
+  };
+}
```

Comment:
To explain this one and use cases other than trigger,

Not all comboboxes will be triggered using a magic character like `/`, `@` or `:`. For instance, someone might want to create a combobox when the user focuses a particular child of a structured node, or when the user presses a keyboard shortcut.

The reason for the `combobox` wrapper was because `query` is a very common option name that doesn't carry much information by itself. I'm forseeing conflicts between the combobox trigger `query` and a more general `query` option on the same plugin.

Perhaps `trigger`, `triggerPreviousCharPattern`, `triggerQuery` and `createComboboxInput`?

Reply:
Makes sense! The new names you suggested seem better.